Generator/Sources/NeedleFramework/Parsing/Pluginized/FileFilters/NonCoreComponentImplFilter.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A filter that performs checks if the file content contains any
+/// non-core component class implementations.
+class NonCoreComponentImplFilter: FileFilter {
+
+    /// Initializer.
+    ///
+    /// - parameter content: The content to be filtered.
+    init(content: String) {
+        self.content = content
+    }
+
+    /// Execute the filter.
+    ///
+    /// - returns: `true` if the file content contains non-core component
+    /// class implementations.
+    func filter() -> Bool {
+        // Use simple string matching first since it's more performant.
+        if !content.contains("NonCoreComponent") {
+            return false
+        }
+
+        // Match actual syntax using Regex.
+        let containsNonCoreComponentInheritance = (Regex(": *(\(needleModuleName).)?NonCoreComponent *<").firstMatch(in: content) != nil)
```

Comment:
Seems like a lot of duplication in the regexes as well: a good opportunity to make them more readable would be to write some convenience inits or static funcs like 
```
static func inheritsFromNeedleModuleName(_ adding: String) -> Regex 
```

Reply:
Good point. I'll consolidate the pluginized regex with the regular regex.